### PR TITLE
Kube vip addons

### PR DESCRIPTION
**Pull request by ocobles**

### CODE REVIEW

**Comment by displague** [comment]
https://github.com/kube-vip/kube-vip/releases/tag/v0.5.11 🎉

File: .utils/generate_addon_structure.sh [hunk 1]
```diff
@@ -173,7 +173,7 @@ EOT
 
 #sanitize vars
 CAPITALIZED_ADDON_NAME=`echo ${ADDON_NAME:0:1} | tr  '[a-z]' '[A-Z]'`${ADDON_NAME:1}
-ADDON_NAME=$(echo $ADDON_NAME | tr '[-]' '_' | tr '[ ]' '_' | tr -dc '[:alnum:]_' | tr '[:upper:]' '[:lower:]')
+ADDON_NAME=$(echo $ADDON_NAME | tr '-' '_' | tr ' ' '_' | tr -dc '[:alnum:]_' | tr '[:upper:]' '[:lower:]')
```
Comment (ctreatma) [inline .utils/generate_addon_structure.sh:176]: Is this a platform-specific change or will this still be OK on non-MacOS machines?

File: modules/kube-vip/examples/simple/main.tf [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+module "equinix_kubernetes_addons" {
+  source = "../../../../"
+
+  # equinix_project = "" // Equinix Metal project ID. Required only if kube_vip_config.cpem_installed is false.
+
+  enable_kube_vip = true
+  # kube_vip_config = {
+  #   # version = "v5.10.0" //if not specified it will get latest available
+  #   # cpem_installed = true //Set true if cloud-provider-equinix-metal addon is being enabled in the module or is already running on the kubernetes cluster. Defaults to true
```
Comment (ctreatma) [inline modules/kube-vip/examples/simple/main.tf]: I think `cpem_installed` is treated as false by default: https://github.com/equinix-labs/terraform-equinix-kubernetes-addons/pull/59/files#diff-ad1372a7935f95af5a8a319d9bbcae337fb64e0eb8ae7dd60ada1f5464348854R23

File: .utils/generate_addon_structure.sh (hunk 1 above)
Comment (ocobles) [reply .utils/generate_addon_structure.sh:176]: Works in linux based OS too. What doesn't work in linux based OS, is checking for both "-" and " " in same tr command `tr '[- ]' '_'`, but when you treat them separately `tr '-' '_' | tr ' ' '_'` it works fine

File: modules/kube-vip/examples/simple/variables.tf [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  type        = string
+}
+
+variable "cpem_installed" {
+  description = <<EOT
+set `false` if cloud-provider-equinix-metal addon is NOT being enabled in the equinix_kubernetes_addons
+module or CPEM is NOT already running on the kubernetes cluster. If not specified it defaults to `true`
+EOT
+  type        = bool
+  default     = false
```
Comment (ctreatma) [inline modules/kube-vip/examples/simple/variables.tf:28]: The description says default is `true`, but the default is set to `false`.  Which is the correct default?

Comment (ocobles) [reply modules/kube-vip/examples/simple/variables.tf:28]: Default is `true` if not specified at all but I want to use `false` for the example. I will update the variable description and add a comment in the main.tf to explain it better

Comment (ocobles) [reply modules/kube-vip/examples/simple/variables.tf:28]: https://github.com/equinix-labs/terraform-equinix-kubernetes-addons/pull/59/commits/73a6715b5a822a3ee7ccddfa3b149e593d485d06

**Review by ctreatma — APPROVED** [review]
I'm on board with this as-is.  I'm not clear if any of the hard-coded values in the kube-vip yaml could be configurable, but if we can & want to support customers putting kube-vip in a different namespace or configuring CPEM with different annotations (or something else not mentioned here), we should add follow-up issues for that.

**Pull request merged**